recipes/opusfile/all/conanfile.py

```diff
@@ -47,30 +67,36 @@ def _build_vs(self):
         includedir = os.path.abspath(os.path.join(self._source_subfolder, "include"))
         with tools.chdir(os.path.join(self._source_subfolder, "win32", "VS2015")):
             msbuild = MSBuild(self)
+            build_type = str(self.settings.build_type)
+            if not self.options.http:
+                build_type += "-NoHTTP"
             msbuild.build_env.include_paths.append(includedir)
             msbuild.build(project_file="opusfile.sln", targets=["opusfile"],
-                          platforms={"x86": "Win32"},
+                          platforms={"x86": "Win32"}, build_type=build_type,
                           upgrade_project=False)
 
     def _configure_autotools(self):
-        if not self._autotools:
-            with tools.chdir(self._source_subfolder):
-                self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
-            args = []
-            if self.options.shared:
-                args.extend(["--disable-static", "--enable-shared"])
-            else:
-                args.extend(["--disable-shared", "--enable-static"])
-            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-            self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        yes_no = lambda v: "yes" if v else "no"
```

Comment:
We might want to have this simple function in conan itself.
AutoToolsBuildEnvironment might be the best place to have it.

Reply:
As long as using the utility function is less hassle.
Copy-pasting/rewriting this helper function is not a lot of work.